src/viur/core/modules/translation.py

```diff
@@ -132,24 +133,29 @@ class TranslationSkel(Skeleton):
 
     public = BooleanBone(
         descr=translate(
-            "core.translationskel.public.descr",
+            "viur.core.translationskel.public.descr",
             "Is this translation public?",
         ),
         defaultValue=False,
     )
 
     @classmethod
-    def write(cls, skelValues: SkeletonInstance, **kwargs) -> db.Key:
-        # Ensure we have only lowercase keys
-        skelValues["tr_key"] = skelValues["tr_key"].lower()
-        return super().write(skelValues, **kwargs)
+    def read(cls, skel, *args, **kwargs):
+        if skel := super().read(skel, *args, **kwargs):
+            if skel["tr_key"]:
+                skel["name"] = skel["tr_key"]
+                skel["tr_key"] = None
+
+        return skel
 
     @classmethod
-    def preProcessSerializedData(cls, skelValues: SkeletonInstance, entity: db.Entity) -> db.Entity:
-        # Backward-compatibility: re-add the key for viur-core < v3.6
-        # TODO: Remove in ViUR4
-        entity["key"] = skelValues["tr_key"]
-        return super().preProcessSerializedData(skelValues, entity)
+    def write(cls, skel, **kwargs):
+
+        # Create the key from the name on initial write!
+        if not skel["key"]:
+            skel["key"] = db.Key(KINDNAME, skel["name"].lower())
```

Comment:
This would overwrite data.

If there's already a complete skeleton with the key `"foo"`, and someone clicks on add and use the same key, it would overwrite the other skel without any warning etc.

Reply:
I didn't tested this, and this is kinda hakkish, but this might solve this issue:

```python
    @classmethod
    def write(cls, skel, **kwargs):

        def write_with_claimed_key():
            skel["key"] = db.Key(KINDNAME, skel["name"].lower())
            if db.Get(skel["key"]):
                raise ValueError(f'An TranslationSkel with {skel["name"]=} exists already!')
            return super().write(skel, **kwargs)

        # Create the key from the name on initial write!
        if not skel["key"]:
            if db.IsInTransaction():
                return write_with_claimed_key()
            else:
                return db.RunInTransaction(write_with_claimed_key)

        return super().write(skel, **kwargs)
```

Reply:
I did some further tests on this: Because `name` is unique, `Skeleton.fromClient()` already reports the problem and the write() is not called. Therefore the behavior is safe here.